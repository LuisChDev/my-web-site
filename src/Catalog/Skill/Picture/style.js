import styled, {css, keyframes} from 'styled-components';
import posed from 'react-pose';
import {cubeColors, cubeImages} from '../../../style.js';
import React from 'react';

/**
 * wrapper for the cube perspective.
 */
const CubeWrapper = styled.div`
  margin-top: 10vh;
  perspective: 1000px;
  perspective-origin: 50% 50%;
`;

/**
 * animation, how the cube should look
 */
const cubeMotion = keyframes`
  from {
    transform: rotateX(0deg) rotateY(0deg);
  }

  to {
    transform: rotateX(360deg) rotateY(360deg);
  }
`;

/**
 * styles for the cube.
 */
const PosedCube = styled.div`
  margin: auto;
  position: relative;
  height: 24vw;
  width: 24vw;
  transform-style: preserve-3d;
  animation: ${cubeMotion} 20s linear infinite;
`;
// const StyledCubeWrap = ({hostRef}) => <StyledCube ref={hostRef}/>;


// /**
//  * container div for the whole cube.
//  */
// const PosedCube = posed(StyledCubeWrap)({
//   rotating: {
//     rotateX: ({rotatex}) => rotatex,
//     rotateY: ({rotatey}) => rotatey,
//   }
// });

/**
 * each individual face of the cube.
 */
const StyledFace = styled.div`
  position: absolute;
  box-sizing: border-box;
  padding: 10px;
  height: 100%;
  width: 100%;
  opacity: 0.8;
  border: solid 1px #eeeeee;
  ${({logo}) => css`
    background-image: url(${cubeImages[logo]});
  `};
  background-size: 150px 150px;
  background-repeat: no-repeat;
  background-position: center;
  ${props => css`
    transform: translate${props.t}(${props.td}vw) rotate${props.r}(${props.rd}deg);
    background-color: ${cubeColors[props.i]};
  `}
`;

export {PosedCube, StyledFace, CubeWrapper};
