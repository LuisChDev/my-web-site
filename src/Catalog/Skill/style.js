import React from 'react';
import styled from 'styled-components';
import posed from 'react-pose';

/**
 * CSS part of the component.
 */
const StyledSkill = styled.div`
  flex-grow: 1;

  display: flex;
  flex-direction: column;
  justify-content: space-between;

  width: 45vw;
  min-height: 20vh;
  font-family: Verdana, Sans-serif;
`;

export default StyledSkill;
