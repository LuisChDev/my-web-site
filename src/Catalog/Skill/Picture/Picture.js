import React, {Component} from 'react';
import {StyledCube, StyledFace, CubeWrapper, CubeImg} from './style.js';

/**
 * this is the component that will be animated with CSS.
 */
const Picture = ({logo, pose}) => {
    return(
      <CubeWrapper>
        <StyledCube pose={pose}>
          {[["X", 0, "Z", 8],
            ["X", 180, "Z", -8],
            ["Y", 90, "X", 8],
            ["Y", -90, "X", -8],
            ["X", 90, "Y", 8],
            ["X", -90, "Y", -8]].map((x,i) =>
                                      <StyledFace r={x[0]} rd={x[1]}
                                                  t={x[2]} td={x[3]}
                                                  i={i} logo={logo}
                                                  key={i}/>)}
        </StyledCube>
      </CubeWrapper>
    );
};

export default Picture;
