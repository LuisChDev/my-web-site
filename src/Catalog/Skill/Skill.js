import React from 'react';
import StyledSkill from './style.js';
import Picture from './Picture/Picture.js';

/**
 * Each individual skill.
 */
const Skill = ({pose, logo, name, description, turns}) => {
  return(
    <StyledSkill>
      <Picture logo={logo} pose={pose} turns={turns}/>
        <div>
          <h2>{name}</h2>
          <p>{description}</p>
        </div>
    </StyledSkill>
  );
};

export default Skill;
