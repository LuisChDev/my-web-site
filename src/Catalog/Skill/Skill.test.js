import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import Skill from './Skill.js';
import Picture from './Picture/Picture.js';
/**
 * Skill is an entirely functional component
 */

describe("Skill component", () => {
  let output;
  beforeEach(() => {
    output = shallow(<Skill
                       name="testName"
                       description="this is a description"
                       pose="testPose"
                       logo="testLogo"/>);
  });
  it("renders a Picture object", () => {
    expect(output.find(Picture).length).toEqual(1);
  });

  it("renders a subtitle and text", () => {
    expect(output.find("h2")).toBeTruthy();
    expect(output.find("p")).toBeTruthy();
  });

  it("renders correctly", () => {
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
