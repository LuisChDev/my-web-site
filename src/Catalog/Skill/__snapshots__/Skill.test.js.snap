// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skill component renders correctly 1`] = `
<styled.div>
  <Picture
    logo="testLogo"
    pose="testPose"
  />
  <div>
    <h2>
      testName
    </h2>
    <p>
      this is a description
    </p>
  </div>
</styled.div>
`;
